mindsdb/integrations/utilities/rag/splitters/file_splitter.py

```diff
@@ -2,80 +2,109 @@
 from typing import Callable, List, Union
 
 from langchain_core.documents import Document
-from langchain_text_splitters import MarkdownHeaderTextSplitter, HTMLHeaderTextSplitter, RecursiveCharacterTextSplitter
+from langchain_text_splitters import (
+    MarkdownHeaderTextSplitter,
+    HTMLHeaderTextSplitter,
+    RecursiveCharacterTextSplitter,
+)
+
+from mindsdb.interfaces.knowledge_base.preprocessing.models import TextChunkingConfig
 
 from mindsdb.utilities import log
 
 DEFAULT_CHUNK_SIZE = 1000
 DEFAULT_CHUNK_OVERLAP = 50
 DEFAULT_MARKDOWN_HEADERS_TO_SPLIT_ON = [
-    ('#', 'Header 1'),
-    ('##', 'Header 2'),
-    ('###', 'Header 3')
+    ("#", "Header 1"),
+    ("##", "Header 2"),
+    ("###", "Header 3"),
 ]
 DEFAULT_HTML_HEADERS_TO_SPLIT_ON = [
-    ('h1', 'Header 1'),
-    ('h2', 'Header 2'),
-    ('h3', 'Header 3'),
-    ('h4', 'Header 4')
+    ("h1", "Header 1"),
+    ("h2", "Header 2"),
+    ("h3", "Header 3"),
+    ("h4", "Header 4"),
 ]
 logger = log.getLogger(__name__)
 
 
 @dataclass
 class FileSplitterConfig:
-    '''Represents configuration needed to split a file into chunks for retrieval.'''
+    """Represents configuration needed to split a file into chunks for retrieval."""
+
     # Target chunk size in characters. Not all splitters will adhere exactly to this (it's more of a guideline)
     chunk_size: int = DEFAULT_CHUNK_SIZE
     # How many characters each chunk should overlap. Not all splitters will adhere exactly to this (it's more of a guideline)
     chunk_overlap: int = DEFAULT_CHUNK_OVERLAP
+    # Chunking parameters are passed as a TextChunkingConfig
+    text_chunking_config: TextChunkingConfig = TextChunkingConfig(
```

Comment:
if we add thus then having chunk_size and chunk_ovlerlap in addition becomes redundant